plugins/scaffolder-backend/src/scaffolder/stages/publish/helpers.ts

```diff
@@ -13,43 +13,57 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Repository, Remote, Signature, Cred } from 'nodegit';
-
-export async function pushToRemoteCred(
-  directory: string,
-  remote: string,
-  credentialsProvider: () => Cred,
-): Promise<void> {
-  const repo = await Repository.init(directory, 0);
-  const index = await repo.refreshIndex();
-  await index.addAll();
-  await index.write();
-  const oid = await index.writeTree();
-  await repo.createCommit(
-    'HEAD',
-    Signature.now('Scaffolder', 'scaffolder@backstage.io'),
-    Signature.now('Scaffolder', 'scaffolder@backstage.io'),
-    'initial commit',
-    oid,
-    [],
-  );
-
-  const remoteRepo = await Remote.create(repo, 'origin', remote);
-
-  await remoteRepo.push(['refs/heads/master:refs/heads/master'], {
-    callbacks: {
-      credentials: credentialsProvider,
-    },
+
+import globby from 'globby';
+import { Logger } from 'winston';
+import { Git } from '@backstage/backend-common';
+
+export async function initRepoAndPush({
+  dir,
+  remoteUrl,
+  auth,
+  logger,
+}: {
+  dir: string;
+  remoteUrl: string;
+  auth: { username: string; password: string };
+  logger: Logger;
+}): Promise<void> {
+  const git = Git.fromAuth({
+    username: auth.username,
+    password: auth.password,
+    logger,
+  });
+
+  await git.init({
+    dir,
+  });
+
+  const paths = await globby(['./**', './**/.*'], {
+    cwd: dir,
+    gitignore: true,
+    dot: true,
   });
-}
 
-export async function pushToRemoteUserPass(
-  directory: string,
-  remote: string,
-  username: string,
-  password: string,
-): Promise<void> {
-  return pushToRemoteCred(directory, remote, () =>
-    Cred.userpassPlaintextNew(username, password),
-  );
+  for (const filepath of paths) {
+    await git.add({ dir, filepath });
+  }
+
+  await git.commit({
+    dir,
+    message: 'Initial commit',
+    author: { name: 'Scaffolder', email: 'scaffolder@backstage.io' },
+    committer: { name: 'Scaffolder', email: 'scaffolder@backstage.io' },
```

Comment:
Could this be `Backstage Scaffolder` or even the actual @backstage-service details (only for GitHub)?

Reply:
Yep, and ideally it would be great to make this configurable too, but just wanted to replicate the current behaviour for now.

Reply:
Actually that being said, for github it would be the app authentication soon too so this would go away for github